20th/articles/rfc43.md

```diff
@@ -0,0 +1,182 @@
+   1050  -18900    79380  -117600   56700
+  -1400   26880  -117600   179200  -88200
+    630  -12600    56700   -88200   44100
+
+Floating Hilbert ⋅ Inverse of Floating Hilbert
+            1             0            0             0            0
+            0             1            0  -7.27596e-12  1.81899e-12
+  2.84217e-14  -6.82121e-13            1  -3.63798e-12            0
+  1.42109e-14  -2.27374e-13  2.72848e-12             1            0
+            0  -2.27374e-13  9.09495e-13  -1.81899e-12            1
+
+```
+
+All those tiny floating point values are "infinitisemal" -- yet some human needs to choose at what point of precision we determine the cutoff. Since we "know" that the inverse dot product is supposed to yield an identity matrix for Hilberts, we can code our algorithm to translate everything below `e-11` into zeroes.
+
+But what about situations that aren't so certain? Some programmers undoubtedly write formal proofs of the safety of using a given cutoff -- however I doubt any claims that this population represents a significant subset of programmers based on lived experience.
+
+### Rational Hilbert
+
+With the rational representation of Hilbert, it's a lot easier to see what is going on with the Hilbert algorithm as the pattern in the rationals clearly evokes the progression. This delivers an ability to "reason backwards" about the numeric data in a way that is not quite as seamless with decimal notation of fractions.[^4]
+
+```
+Rational Hilbert
+    1  1/2  1/3  1/4  1/5
+  1/2  1/3  1/4  1/5  1/6
+  1/3  1/4  1/5  1/6  1/7
+  1/4  1/5  1/6  1/7  1/8
+  1/5  1/6  1/7  1/8  1/9
+
+Inverse (same output, trimmed here for space)
+
+Rational Hilbert ⋅ Inverse of Rational Hilbert
+  1  0  0  0  0
+  0  1  0  0  0
+  0  0  1  0  0
+  0  0  0  1  0
+  0  0  0  0  1
+```
+
+Notice the lack of ambiguity in both the initial Hilbert data set and the final output of the inverse dot product. There is no need to choose any threshold values, thus no programmer decisions come in between the result of the equation provided by the computer and the result of the equation as stored by the computer or presented to the user.
+
+Consider again the fact that it is useless to do an equality comparison in floating point math -- the imprecision of the underlying representation forces users to choose some decimal place to end the number they check against. There is no possible model that can take into account the total degree of impact this human interaction (in aggregate) creates on top of the known and quantifiable imprecisions that are usually the basis of the "it's good enough" argument.
+
+## The cost of precision
+
+In summary, the nice equations about maximumum loss per equation are far from the entire story and it is in recognition of that fact that Raku does it's best to ensure that the eventual user of whatever Raku-powered system is not adversely impacted by unintentional or negligent errors in the underlying math used by the programmer.
```

Comment:
"nice equations about maximumum loss per equation" maybe you mean *formulas*? Also, one too many "mu"s.